# Bulk Segregant Approaches to Nutritional Genomics in Plasmodium falciparum

Sudhir Kumar1¶, Xue Li2¶, Marina McDew-White2, Ann Reyes2, Abeer Sayeed2, Meseret T. Haile1, Spencer Y. Kennedy1, Nelly M. Camargo1, Lisa A. Checkley3, Katelyn M. Vendrely3, Katrina A. Button-Simons3, Ian H. Cheeseman4, Stefan H. I. Kappe1,5, François Nosten,6,7, Michael T. Ferdig3, Ashley M. Vaughan1,5* and Tim J. C. Anderson2*

1 Center for Global Infectious Disease Research, Seattle Children’s Research Institute, Seattle, Washington, USA.
2 Program in Disease Intervention and Prevention, Texas Biomedical Research Institute, San Antonio, Texas, USA.
3 Eck Institute for Global Health, Department of Biological Sciences, University of Notre Dame, Notre Dame, Indiana, USA.
4 Program in Host Pathogen Interactions, Texas Biomedical Research Institute, San Antonio, Texas, USA.
5 Department of Global Health, University of Washington, Seattle, Washington, USA.
6 Shoklo Malaria Research Unit, Mahidol-Oxford Tropical Medicine Research Unit, Faculty of Tropical Medicine, Mahidol University, Mae Sot, Thailand.
7 Centre for Tropical Medicine and Global Health, Nuffield Department of Medicine Research building, University of Oxford Old Road campus, Oxford, UK.

¶ Sudhir Kumar and Xue Li contributed equally to this work.
*Co-corresponding authors: ashley.vaughan@seattlechildrens.org (AMV) and tanderso@TxBiomed.org (TJCA).


Nutrient acquisition/metabolism pathways provide potent targets for drug design. We conducted crosses between African (NF54) and Asian (NHP4026) malaria parasites, and compared genome-wide allele frequency changes in independent progeny populations grown in human serum or AlbuMAX, a commercial bovine serum formulation. We detected three QTLs linked with differential growth that contained strong candidate genes: aspartate transaminase AST (chromosome 2), cysteine protease ATG4 (chr. 13) and EBA-140 (chr. 14). Alleles inherited from NF54 (chr. 2 and 14) and from NHP4026 (chr. 13) were positively selected in AlbuMAX, while the same alleles were selected against in serum. Selection driving differential growth was strong (s = 0.10 – 0.23 per 48-hour lifecycle) and observed in all biological replicates. These results demonstrate the effectiveness of bulk segregant approaches for revealing nutritional polymorphisms in Plasmodium falciparum. This approach will allow systematic dissection of nutrient acquisition/metabolism pathways that are potential targets for intervention against P. falciparum.
