# A malaria parasite cross reveals genetic determinants of Plasmodium falciparum growth in different culture media

Sudhir Kumar1†, Xue Li2†, Marina McDew-White2, Ann Reyes2, Elizabeth Delgado2, Abeer Sayeed2, Meseret T. Haile1, Biley A. Abatiyow1, Spencer Y. Kennedy1, Nelly M. Camargo1, Lisa A. Checkley3, Katelyn V. Brenneman3, Katrina A. Button-Simons3, Manoj T. Duraisingh4, Ian H. Cheeseman5, Stefan H. I. Kappe1,6, François Nosten,7,8, Michael T. Ferdig3, Ashley M. Vaughan1,6* and Tim J. C. Anderson2*

1 Center for Global Infectious Disease Research, Seattle Children’s Research Institute, Seattle, Washington, USA.
2 Program in Disease Intervention and Prevention, Texas Biomedical Research Institute, San Antonio, Texas, USA.
3 Eck Institute for Global Health, Department of Biological Sciences, University of Notre Dame, Notre Dame, Indiana, USA.
4Immunology & Infectious Diseases Department, Harvard T.H. Chan School of Public Health, Boston, MA, USA.
5 Program in Host Pathogen Interactions, Texas Biomedical Research Institute, San Antonio, Texas, USA.
6 Department of Pediatrics, University of Washington, Seattle, Washington, USA.
7 Shoklo Malaria Research Unit, Mahidol-Oxford Tropical Medicine Research Unit, Faculty of Tropical Medicine, Mahidol University, Mae Sot, Thailand.
8 Centre for Tropical Medicine and Global Health, Nuffield Department of Medicine Research building, University of Oxford Old Road campus, Oxford, UK.
† Sudhir Kumar and Xue Li contributed equally to this work.
*Co-corresponding authors: ashley.vaughan@seattlechildrens.org (AMV) and tanderso@TxBiomed.org (TJCA).

What genes determine in vitro growth and nutrient utilization in asexual blood-stage malaria parasites? Competition experiments between 3D7, a lab-adapted African parasite, and a recently isolated Asian parasite (NHP4026) reveal contrasting outcomes in different media: 3D7 outcompetes NHP4026 in media containing human serum, while NHP4026 outcompetes 3D7 in media containing AlbuMAX, a commercial lipid-rich bovine serum formulation. To determine the basis for this polymorphism, we conducted parasite genetic crosses using humanized mice and compared genome-wide allele frequency changes in three independent progeny populations cultured in media containing human serum or AlbuMAX. This bulk segregant analysis detected three quantitative trait loci (QTL) regions (on Chr 2 containing aspartate transaminase AST; chr 13 containing EBA-140; and chr 14 containing cysteine protease ATG4) linked with differential growth in serum or AlbuMAX in each of three independent progeny pools. Selection driving differential growth was strong (s = 0.10 – 0.23 per 48-hour lifecycle).  We conducted validation experiments for the strongest QTL on chr 13: competition experiments between ∆EBA-140 and 3D7 wildtype parasites showed fitness reversals in the two medium types as seen in the parental parasites, validating this locus as the causative gene. These results (i) demonstrate the effectiveness of bulk segregant analysis for dissecting fitness traits in P. falciparum genetic crosses, and (ii) reveal intimate links between red blood cell invasion and nutrient composition of growth media.  Use of parasite crosses combined with bulk segregant analysis will allow systematic dissection of key nutrient acquisition/metabolism and red blood cell invasion pathways in P. falciparum. 

# https://www.frontiersin.org/articles/10.3389/fcimb.2022.878496/full
